sdk/java/pulumi/src/main/java/com/pulumi/test/PulumiTest.java

```diff
@@ -10,12 +14,127 @@
+     *
+     * @param stack the stack to run in Pulumi test runtime
+     * @return a {@link TestResult} from Pulumi test runtime after running the stack
+     * @see #runTestAsync(Consumer)
+     * @see #withOptions(TestOptions)
+     * @see #withMocks(Mocks)
+     */
+    static TestResult runTest(Consumer<Context> stack) {
+        return withOptions(TestOptions.Empty).runTest(stack);
+    }
+
     /**
      * Run a Pulumi test stack callback asynchronously and return {@link TestResult}.
      *
      * @param stack the stack to run in Pulumi test runtime
      * @return a future {@link TestResult} from Pulumi test runtime after running the stack
+     * @see #runTest(Consumer)
+     * @see #withOptions(TestOptions)
+     * @see #withMocks(Mocks)
+     */
+    static CompletableFuture<TestResult> runTestAsync(Consumer<Context> stack) {
+        return withOptions(TestOptions.Empty).runTestAsync(stack);
+    }
+
+    /**
+     * Use {@link TestOptions} in this test.
+     *
+     * @param options the {@link TestOptions} to use
+     * @return a Pulumi test program entrypoint with given {@link TestOptions}
+     * @see #runTest(Consumer)
+     * @see #runTestAsync(Consumer)
+     * @see #withMocks(Mocks)
+     */
+    static PulumiTest.API withOptions(TestOptions options) {
+        return new PulumiTestInternal.APIInternal().withOptions(options);
+    }
+
+    /**
+     * Use {@link Mocks} in this test.
+     *
+     * @param mocks the {@link com.pulumi.deployment.internal.Monitor} mocks to use
+     * @return a Pulumi test program entrypoint with given {@link TestOptions}
+     * @see #runTest(Consumer)
+     * @see #runTestAsync(Consumer)
+     * @see #withOptions(TestOptions)
+     */
+    static PulumiTest.API withMocks(Mocks mocks) {
+        return new PulumiTestInternal.APIInternal().withMocks(mocks);
+    }
+
+    /**
+     * Cleanup internal test state. <b>Must be called</b> after every run of a Pulumi test program.
+     */
+    static void cleanup() {
+        PulumiTestInternal.cleanup();
+    }
+
+    /**
+     * Extract the value from an {@link Output} blocking.
+     *
+     * @param output the {@link Output} to extract value from
+     * @return the underlying value of the given {@link Output}
      */
-    CompletableFuture<TestResult> runTestAsync(Consumer<Context> stack);
+    @Nullable
+    static <T> T extractValue(Output<T> output) {
+        return extractValueAsync(output).join();
+    }
+
+    /**
+     * Extract the future value from an {@link Output} asynchronously.
+     *
+     * @param output the {@link Output} to extract future value from
+     * @return the underlying future value of the given {@link Output}
+     */
+    static <T> CompletableFuture<T> extractValueAsync(Output<T> output) {
+        return Internal.of(output).getValueNullable();
+    }
+
+    /**
+     * Pulumi test entrypoint operations.
+     */
+    interface API {
```

Comment:
I was thinking about renaming this interface to `DSL` (here and in `Pulumi` ofc), WDYT?

Reply:
Nope, it's an abbreviation some users might not know. The current name is not great but let's not change names unless there's a clear win.